CLIENT-SPECIFICATION.md

```diff
@@ -116,7 +116,12 @@ The structure inside these translation folders is identical to that of the main
 
 ## Page structure
 
-Although this specification is about the interface that clients must provide, it is also worth noting that pages are written in standard [CommonMark](https://commonmark.org/), which the exception of the non-standard `{{` and `}}` syntax, which surrounds values in an example that users may edit. Clients MUST NOT break if the page format is changed within the _CommonMark_ specification.
+Although this specification is about the interface that clients must provide, it is also worth noting that pages are written in standard [CommonMark](https://commonmark.org/), with the exception of the non-standard `{{` and `}}` placeholder syntax, which surrounds values in an example that users may edit. Clients MAY highlight the placeholders and MUST remove the surrounding curly braces. Clients MUST display the `{{` and `}}` if escaped using `\` (i.e. `\{\{` and `\}\}`). Clients MUST NOT break if the page format is changed within the _CommonMark_ specification.
+
+### Examples
+
+- `ping {{example.com}}` MUST be rendered as "ping example.com"
+- `docker inspect --format '\{\{range.NetworkSettings.Networks\}\}\{\{.IPAddress\}\}\{\{end\}\}' {{container}}` MUST be rendered as "docker inspect --format '{{range.NetworkSettings.Networks}}{{.IPAddress}}{{end}}' container"
```

Comment:
I would personally lighten the syntax to only one backslash:
`docker inspect --format '\{{range.NetworkSettings.Networks\}}\{{.IPAddress\}}\{{end\}}' {{container}}`.

This is maybe easier to parse than what I had proposed, but still not too heavy for reading source pages (even though this isn't done often). The drawback could if we needed to render something like `\{{ <...> }}`

But since I've seen only 3 occurrences of this problem, maybe this change is not needed. 🤷🏻

Reply:
I favor a backslash for each curly brace. I think it's much more intuitive to treat things per character, rather than by our special definition attributed to combinations of characters.

In Markdown, `\{` already means a literal `{`. So we should run with it. I suspect this will also be more backward compatible. When we don't mean the placeholder syntax, `{{` is never found, so even if the client doesn't know to escape it, it'll inadvertently ignore it anyway.

I believe this will be simpler than giving `{{` conditional meaning based on an escape sequence multiple characters away.

| Syntax | Arg (curly braces removed) | Rendered |
|---|---|---|
| `{{uwu}}` | `uwu` | uwu |
| `{{\{uwu\}}}` | `\{uwu\}` | \{uwu\} |
| `{{\{\{uwu\}\}}}` | `\{\{uwu\}\}` | \{\{uwu\}\} |

The main difference, here, is that having a single escape sequence affect multiple characters would require special handling from clients, while doing one escape sequence per character is already supported/expected by clients and CommonMark parsers.